accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,530 @@
+
+#### Comparison of provided and required ID token claims
+
+The following claims are required to fully support generating build provenance for npm releases.
+
+| Claim           | Required                                                                  | GitHub           | GitLab                                  | CircleCI                                                       |
+|-----------------|---------------------------------------------------------------------------|------------------|-----------------------------------------|----------------------------------------------------------------|
+| Subject         | The repo/project/resource from where the build is running.                | Repository path  | Project path                            | “org/org id  project/project id user/user id”                  |
+| Repo            | The git/scm repository from where the build is running.                   | repository       | project_path                            | Only contains the project id, which can be seen as a weak link |
+| Repo id         | The ID of the git/scm repository from where the build is running.         | repostory_id     | project_id                              | See above comment on “Repo”                                    |
+| Repo ref        | The git/scm reference that triggered the build.                           | ref              | ref                                     | N/A                                                            |
+| Commit SHA      | The git/scm sha that triggered the build.                                 | sha              | N/A                                     | N/A                                                            |
+| Actor           | The account that initiated the build.                                     | actor            | user_login or user_email                | Need to parse subject claim.                                   |
+| Actor id        | The ID of the account that initiated the build.                           | actor_id         | user_id                                 | See above comment on “Actor”                                   |
+| Build name      | The name of the build.                                                    | workflow         | pipeline_id (i.e no name, only numeric) | N/A                                                            |
+| Build id        | The ID of the build.                                                      | run_id + run_attempt           |                                         |                                                                |
+| Build reference | Ref to an external workflow if one is being used, e.g. a trusted builder. | job_workflow_ref | N/A                                     | N/A                                                            |
+
+## Implementation details
```

Comment:
I find this section misleading because there are actually **two** different modes to publish, but that fact is not obvious (at least to me):

1. The "sigstore integration in the npm CLI", i.e. the CLI does the signing by grabbing an OIDC token from the CI/CD system.
2. The "Non-falsifiable provenance using a trusted builder", i.e. the CI/CD does the signing with no integration with the CLI.

I suggest calling this out up front and perhaps having a diagram illustrating the two different models. The former has weaker security guarantees around the accuracy of the provenance.

Reply:
@MarkLodato yes this is a great callout. I'll clarify this up front 👍